python/cudf_polars/cudf_polars/dsl/expressions/boolean.py

```diff
@@ -94,6 +97,15 @@ def __init__(
             # TODO: If polars IR doesn't put the casts in, we need to
             # mimic the supertype promotion rules.
             raise NotImplementedError("IsIn doesn't support supertype casting")
+        if self.name is BooleanFunction.Name.IsIn:
+            _, haystack = self.children
+            # TODO: Use pl.List isinstance check once we have https://github.com/rapidsai/cudf/pull/18564
+            if isinstance(haystack, LiteralColumn) and isinstance(
+                haystack.value, pa.ListArray
```

Comment:
CC @mroeschke (for awareness)

If this PR gets in before yours https://github.com/rapidsai/cudf/pull/18564

Reply:
No because the list is nested